src/ex_docmd.c

```diff
@@ -2904,10 +2904,12 @@ do_one_cmd(
 #ifdef FEAT_LISTCMDS
     /*
      * Accept buffer name.  Cannot be used at the same time with a buffer
-     * number.  Don't do this for a user command.
+     * number.  Don't do this for a user command or :bvimgrep[add]
      */
     if ((ea.argt & BUFNAME) && *ea.arg != NUL && ea.addr_count == 0
-	    && !IS_USER_CMDIDX(ea.cmdidx))
+	    && !IS_USER_CMDIDX(ea.cmdidx) && ea.cmdidx != CMD_bvimgrep
+	    && ea.cmdidx != CMD_bvimgrepadd && ea.cmdidx != CMD_blvimgrep
+	    && ea.cmdidx != CMD_blvimgrepadd)
```

Comment:
If the `-b` suggestion were implemented, these special cases sprinkled everywhere could be avoided.